crates/settlement-clients/starknet/src/tests/test.rs

```diff
@@ -0,0 +1,229 @@
+    duration: Duration,
+) -> bool {
+    let mut attempt = 0;
+    loop {
+        attempt += 1;
+        if attempt >= 5 {
+            return false;
+        }
+        let reciept = match account.provider().get_transaction_status(transaction_hash).await {
+            Ok(reciept) => reciept,
+            Err(ProviderError::StarknetError(StarknetError::TransactionHashNotFound)) => {
+                tokio::time::sleep(duration).await;
+                continue;
+            }
+            _ => panic!("Unknown error"),
+        };
+
+        match reciept {
+            TransactionStatus::Received => (),
+            TransactionStatus::Rejected => return false,
+            TransactionStatus::AcceptedOnL2(status) => match status {
+                TransactionExecutionStatus::Succeeded => return true,
+                TransactionExecutionStatus::Reverted => return false,
+            },
+            TransactionStatus::AcceptedOnL1(status) => match status {
+                TransactionExecutionStatus::Succeeded => return true,
+                TransactionExecutionStatus::Reverted => return false,
+            },
+        }
+        // This is done, since currently madara does not increment nonce for pending transactions
+        tokio::time::sleep(duration).await;
+    }
+}
+
+#[fixture]
+async fn setup() -> (SingleOwnerAccount<JsonRpcClient<HttpTransport>, LocalWallet>, MadaraCmd) {
+    let _ = env_logger::builder().is_test(true).try_init();
+    dotenvy::from_filename_override(".env.test").expect("Failed to load the .env file");
+
+    let madara_process = spin_up_madara().await;
+    env::set_var("STARKNET_RPC_URL", madara_process.rpc_url.to_string());
+    println!("RPC url {:?}", madara_process.rpc_url);
+
+    let env_settings = EnvSettingsProvider::default();
+    let rpc_url = Url::parse(&env_settings.get_settings_or_panic("STARKNET_RPC_URL")).unwrap();
+
+    // let provider = JsonRpcClient::new(HttpTransport::new(Url::parse("http://localhost:9944").unwrap()));
+    let provider: JsonRpcClient<HttpTransport> = JsonRpcClient::new(HttpTransport::new(rpc_url));
+    let signer = LocalWallet::from(SigningKey::from_secret_scalar(
+        Felt::from_hex(&env_settings.get_settings_or_panic("STARKNET_PRIVATE_KEY")).expect("Invalid private key"),
+    ));
+    let address = Felt::from_hex(&env_settings.get_settings_or_panic("STARKNET_PUBLIC_KEY")).unwrap();
+
+    let chain_id = provider.chain_id().await.unwrap();
+    let mut account = SingleOwnerAccount::new(provider, signer, address, chain_id, ExecutionEncoding::New);
+
+    // `SingleOwnerAccount` defaults to checking nonce and estimating fees against the latest
+    // block. Optionally change the target block to pending with the following line:
+    account.set_block_id(BlockId::Tag(BlockTag::Pending));
+    (account, madara_process)
+}
+
+#[rstest]
+#[tokio::test]
+async fn test_deployment(#[future] setup: (SingleOwnerAccount<JsonRpcClient<HttpTransport>, LocalWallet>, MadaraCmd)) {
```

Comment:
can you create a type alias for `SingleOwnerAccount<JsonRpcClient<HttpTransport>, LocalWallet>` for easier readability

Reply:
Created